src/App.js

```diff
@@ -1,24 +1,61 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'bulma/css/bulma.css';
+import foodsArr from './foods.json';
+import FoodBox from './components/FoodBox';
+import { useState } from "react"; 
+import FoodForm from './components/FoodForm';
+import Search from './components/Search';
 
 function App() {
+
+  const [foods, setFoods] = useState(foodsArr);
+  /* console.log(foods[0].name) */
+
+  const [showForm, setShowForm] = useState(false);
+
+  const createNewFood = (newFood) => {
+    setFoods((prevFood)=>{
+      return [newFood, ...prevFood];
+    });
+  }
+
+  const searchFood = (search) => {
+    const searchResults = foods.filter((food) => food.name.includes(search))
+    console.log(searchResults)
+    setFoods(searchResults);
+  }
+  
+  const hideForm = () => {
+    setShowForm(!showForm)
+  }
+  
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <h1>IronNutrition</h1>
+     
+    <br/>
+    <Search searchFood={searchFood} />  
+    <br/>
+
+    <button className='button-hide' onClick={hideForm}>{showForm ? "Hide form" : "Add new food"}</button>
+    {showForm && <FoodForm createFood= {createNewFood}/>} 
+    <br/>
+    <br/>
+    <br/>
+   
+    {foods.map((element, index) => {
```

Comment:
Instead of displayig original array, we want to always display filtered array (if there is no search input -> we still have initial value from json)

  {filteredFoods.map((element, index) => {}